android/app/build.gradle

```diff
@@ -152,22 +227,25 @@ android {
             reset()
             enable enableSeparateBuildPerCPUArchitecture
             universalApk enableSeparateBuildPerCPUArchitecture  // If true, also generate a universal APK
-            include "armeabi-v7a", "x86", "arm64-v8a", "x86_64"
+            include (*reactNativeArchitectures())
         }
     }
     buildTypes {
+        def useReleaseKey = project.hasProperty('MATTERMOST_RELEASE_STORE_FILE')
         release {
             minifyEnabled enableProguardInReleaseBuilds
             proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
-            signingConfig signingConfigs.release
+            if (useReleaseKey) {
+                signingConfig signingConfigs.release
+            } else {
+                signingConfig signingConfigs.debug
+            }
         }
         debug {
-            minifyEnabled enableProguardInReleaseBuilds
-            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
-            if (nativeArchitectures) {
-                ndk {
-                    abiFilters nativeArchitectures.split(',')
-                }
+            if (useReleaseKey) {
```

Comment:
so that if you run `npm run android -- --variant=release` and you have release keys set it uses the same ones to avoid adb complaining on not being able to install the app on the device.

this was done for v2 a few months ago